forte/processors/base/data_augment_processor.py

```diff
@@ -46,44 +55,559 @@ class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
     levels: character, word, sentence or document.
     """
     def __init__(self):
-        """
-        The replaced_spans records the entries replaced by new texts.
-        It consists of tuples (entry, new text) inserted by :func: replace.
-        The new text will be used for building new data pack.
+        r"""
+        - replaced_annos: {datapack id: List[span, text]}
+            It records the spans replaced by new texts.
+            It is a map from datapack id to a list of tuples
+            (span, new text) inserted by :func: replace.
+            The new text will be used for building new data pack.
+
+        - inserted_annos: {datapack id: Dict{position: length}}
+            It records the inserted spans, mapping from datapack id
+            to a dictionary (position -> length) inserted
+            by :func: insert.
+            The position is the index in the original datapack
+            of insertion, and the length is the length of the
+            inserted string.
+
+        - deleted_annos: {datapack id: Set[annotation tid]}
+            It records the deleted spans, mapping from datapack id
+            to a set of annotation tids appended by :func: delete.
+
+        - data_pack_map: {orig pack id: new pack id}
+            It maintains a mapping from the pack id
+            of the original pack to the pack id of augmented pack.
+            It is used when copying the MultiPackLink.
+
+        - anno_maps: {datapack id: Dict{orig tid, new tid}}
+            It is a map for tracking the annotation ids
+            before and after the auto align. It maps the
+            original annotation tid to the new annotation tid.
+            It is used when copying the MultiPackLink.
         """
         super().__init__()
-        self.replaced_spans: List[Tuple[Entry, str]] = []
+        self.replaced_annos: DefaultDict[int, List[Tuple[Span, str]]] = \
```

Comment:
Most of the variables should be private (or at least protected). Please rename with leading `__`